File: native/android/AndroidAwnCore/src/main/kotlin/me/carda/android_awn_core/StringUtils.kt
package me.carda.android_awn_core

/**
 * Instance-based string helpers (singleton via getInstance, instance methods —
 * no static utilities), mirroring the original AndroidAwnCore `StringUtils`.
 */
class StringUtils private constructor() {
    companion object {
        private var instance: StringUtils? = null
        fun getInstance(): StringUtils = instance ?: StringUtils().also { instance = it }
    }

    fun isNullOrEmpty(value: String?): Boolean = value == null || value.trim().isEmpty()
}

File: native/apple/Sources/IosAwnCore/StringUtils.swift
import Foundation

/// Instance-based string helpers (singleton, instance methods — no static
/// utilities), mirroring the original IosAwnCore `StringUtils`.
public final class StringUtils {
    public static let shared = StringUtils()
    private init() {}

    public func isNullOrEmpty(_ value: String?) -> Bool {
        guard let value = value else { return true }
        return value.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
}
